repository/models.go

```diff
@@ -1,24 +1,191 @@
 package repository
 
 import (
+	"attendance/util"
+	"encoding/json"
+	"net/http"
+	"os"
 	"time"
 
-	"github.com/google/uuid"
+	"github.com/go-pg/pg"
+	"go.uber.org/zap"
 )
 
 type User struct {
-	UserID   uuid.UUID
 	Username string `json:"username"`
-	Password string `json:"password"`
+	Password string `json:"password,omitempty"`
 	FullName string `json:"fullname"`
 	Class    int    `json:"class"`
-	Email    string `json:"email"`
+	Email    string `json:"email,omitempty"`
 	Role     string `json:"role"`
 }
 
 type Attendance struct {
-	UserID       uuid.UUID
-	AttendanceID uuid.UUID
-	PunchInDate  time.Time
-	PunchOutDate time.Time
+	Username     string    `pg:"username" json:"-"`
+	AttendanceID string    `pg:"attendance_id,pk" json:"-"`
+	PunchInDate  time.Time `pg:"punch_in_date"`
+	PunchOutDate time.Time `pg:"punch_out_date"`
+}
+
+type AttendanceJSON struct {
+	PunchInDate  time.Time `pg:"punch_in_date"`
+	PunchOutDate time.Time `pg:"punch_out_date"`
+}
+
+type GetTeacherAttendanceJSON struct {
+	ID    string `json:"id"`
+	Month int    `json:"month"`
+	Year  int    `json:"year"`
+}
+
+type StudentAttendanceJSON struct {
```

Comment:
this is in directory repository ideally it should contain only repository models, other struct can be moved outside to common place